### PR TITLE
MGMT-2392 - allow updating api vip dnsname/ip in day2 cluster

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -567,6 +574,19 @@ func (b *bareMetalInventory) formatNodeIgnitionFile(apiVipDnsname string) ([]byt
 	return buf.Bytes(), nil
 }
 
+func (b *bareMetalInventory) createAndUploadNodeIgnition(ctx context.Context, clusterID *strfmt.UUID, apiVipDnsname string) error {
+	ignitionString, err := b.formatNodeIgnitionFile(apiVipDnsname)
+	if err != nil {
+		return errors.Errorf("Failed to create ignition string for cluster %s", clusterID)
+	}
+	fileName := fmt.Sprintf("%s/%s", clusterID, workerIgnition)
+	err = b.objectHandler.Upload(ctx, ignitionString, fileName)
```
Comment (romfreiman) [inline internal/bminventory/inventory.go:583]: log?

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:583]: the log on line 1270 will output the appropriate error

Comment (romfreiman) [reply internal/bminventory/inventory.go:583]: and it all goes well? Dont we want to know it happened :)

Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:583]: fixed

File: internal/bminventory/inventory_test.go [hunk 2]
```diff
@@ -1516,6 +1516,33 @@ var _ = Describe("cluster", func() {
...
 
+			Context("Day2 api vip dnsname/ip", func() {
+				BeforeEach(func() {
+					clusterID = strfmt.UUID(uuid.New().String())
+					err := db.Create(&common.Cluster{Cluster: models.Cluster{
+						ID:   &clusterID,
+						Kind: swag.String(models.ClusterKindAddHostsCluster),
+					}}).Error
+					Expect(err).ShouldNot(HaveOccurred())
+					//addHost(masterHostId1, models.HostRoleMaster, "known", clusterID, getInventoryStr("1.2.3.4/24", "10.11.50.90/16"), db)
```
Comment (filanov) [inline internal/bminventory/inventory_test.go:1527]: please remove if not used.

File: internal/bminventory/inventory_test.go [hunk 3]
```diff
@@ -1516,6 +1516,33 @@ var _ = Describe("cluster", func() {
 				})
 			})
 
+			Context("Day2 api vip dnsname/ip", func() {
```
Comment (filanov) [inline internal/bminventory/inventory_test.go:1519]: how about testing `RegisterAddHostsCluster`?

File: internal/bminventory/inventory_test.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory_test.go:1527]: fixed

File: internal/bminventory/inventory_test.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory_test.go:1519]: there is already a test that does it. the function was just re-formatted

**Review by filanov — APPROVED** [review]

**Pull request merged**